lxd/storage/drivers/driver_zfs_volumes.go

```diff
@@ -2653,7 +2648,10 @@ func (d *zfs) MountVolumeSnapshot(snapVol Volume, op *operations.Operation) erro
 		// Order is important here, the parent volmode=dev must be set before snapdev=visible otherwise
 		// it won't take effect.
 		if parentVolMode != "dev" {
-			return fmt.Errorf("Parent block volume needs to be mounted first")
```

Comment:
I'd like to understand why this is needed, this looks like a sanity check that is being removed, and the expectation is that the caller has already mounted the parent volume (as was happening above) which would have managed the ref counters .